src/types.ts

```diff
@@ -63,26 +63,24 @@ export interface ProtocolWithReturn<Data, Return> {
  * Extendable by user.
  */
 export interface ProtocolMap {
-  // foo: { id: number, name: string }
+  __crx_bridge_stream_open__: ProtocolWithReturn<{ channel: string; streamId: string }, boolean>
+  __crx_bridge_stream_transfer__: { streamId: string; action: 'transfer' | 'close'; streamTransfer: JsonValue }
+  // foo: { id: number; name: string; }
   // bar: ProtocolWithReturn<string, number>
 }
 
 export type DataTypeKey = keyof ProtocolMap
 
+type ValidateJsonValue<T> = T extends JsonValue ? T : 'This value is not a serializable JSON value'
```

Comment:
btw. `JsonValue` is probably a little bit too restrictive - for instance, it doesn't allow `undefined` but that's definitely transferrable. I didn't verify it but if this `postMessage` API works exactly the same as the one on windows then even more stuff can be actually sent over it. The traditional `postMessage` is using `structuredClone` algorithm under the hood. It might be hard to replicate its requirements on the type-level though (it might be possible, but I question if it should be done). The supported types are listed [here](https://developer.mozilla.org/en-US/docs/Web/API/Web_Workers_API/Structured_clone_algorithm#supported_types) but there are also gotchas there that might not be capturable at the type-level anyway.

Should we drop this validation altogether?